src/main/kotlin/me/owdding/customscoreboard/feature/customscoreboard/ScoreboardLine.kt

```diff
@@ -27,6 +36,39 @@ data class ScoreboardLine(
         isBlank,
     )
 
+    private var actions: Map<Element.Actions, Any> = emptyMap()
+
+    val widget by lazy {
+        Widgets.button {
+            it.setSize(layout.width, layout.height)
+            it.withTexture(null)
+            it.withRenderer { graphics, ctx, ticks ->
+                graphics.translated(it.x, it.y) {
+                    layout.render(graphics, ctx.mouseX, ctx.mouseY, ticks)
+                }
+            }
+
+            if (Element.Actions.HOVER in actions) {
+                it.withTooltip(Text.multiline(actions[Element.Actions.HOVER]))
+                it.setTooltipDelay(Duration.of(-1, ChronoUnit.SECONDS))
+            }
+
+            it.withCallback {
+                actions.forEach { (k, v) ->
+                    when (k) {
+                        Element.Actions.CLIENT_COMMAND -> McClient.self.connection?.sendCommand((v as String).removePrefix("/"))
+                        Element.Actions.SERVER_COMMAND -> McClient.sendCommand((v as String).removePrefix("/"))
```

Comment:
both of these can just be the same as the connection one if it cant be done on client will just try sending to the server